### PR TITLE
fix deleted queries not removed from side menu

### PR DESCRIPTION
**Pull request by 0xF013**
[`* `#17231` deleted queries not removed from side menu`](http://community.openproject.org/work_packages/17231)
- [ ]  fix intendation
- [ ] rebase againt release/4.0

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://0xf013-fix-17231-deleted-queries-not-removed-from-sidemenu.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://0xf013-fix-17231-deleted-queries-not-removed-from-sidemenu.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by 0xF013** [comment]
I think it's flickering. I tried the test locally and it was gren

File: frontend/tests/unit/tests/services/query-service-test.js [hunk 1]
```diff
@@ -85,6 +91,35 @@ describe('QueryService', function() {
...
+        ).respond(200, []);
+        $httpBackend.when(
+          'GET',
+          PathHelper.apiProjectGroupedQueriesPath(1)
+        ).respond(200, []);
+        QueryService.deleteQuery();
+        $httpBackend.flush();
+    });
+
+  afterEach(function() {
```
Comment (myabc) [inline frontend/tests/unit/tests/services/query-service-test.js:114]: @0xF013 can you fix the indent?

**Comment by ulferts** [comment]
:+1:

**Comment by ulferts** [comment]
works as advertised

File: frontend/tests/unit/tests/services/query-service-test.js (hunk 1 above)
Comment (0xF013) [inline frontend/tests/unit/tests/services/query-service-test.js:114]: @myabc sure, never noticed :-\

**Comment by ulferts** [comment]
@0xF013 another thing that occurred to me: This is supposed to be issued against release/4.0 branch. Does it match?

**Comment by 0xF013** [comment]
@ulferts will rebase

**Comment by myabc** [comment]
@0xF013 you'll need to close/re-open PR.

**Comment by 0xF013** [comment]
@myabc shouldn't it be merged into dev as well?

**Comment by myabc** [comment]
@0xF013 it will be. `release/4.0` is continually merged back into dev.

**Comment by 0xF013** [comment]
@myabc I see, will do

**Comment by myabc** [comment]
Closed in favour of #2443 (against `release/4.0`)

**Pull request closed without merging**